tests/Doctrine/Tests/DBAL/Functional/Schema/SQLServerSchemaManagerTest.php

```diff
@@ -404,4 +405,30 @@ public function testPkOrdering(): void
         self::assertEquals('colB', $columns[0]);
         self::assertEquals('colA', $columns[1]);
     }
+
+    public function testPartialIndexes(): void
+    {
+        if (! $this->schemaManager->getDatabasePlatform()->supportsPartialIndexes()) {
```

Comment:
Would it be possible to move this test one level higher, to the `SchemaManagerFunctionalTestCase` and reuse for all platforms that support partial indexes?

Since it's a functional test, it should cover the API on all platforms that declare support for it, and we already have one in https://github.com/doctrine/dbal/blob/3f3551aadb20c7ed58ee9c7f06ed74770588272f/tests/Doctrine/Tests/DBAL/Functional/Schema/PostgreSqlSchemaManagerTest.php#L355

Reply:
No, not at the moment, because of the limitation/bug #3871 - Postgres and SQL Server automatically add (different) brackets, type hints etc., so the returned where condition depends on the platform (for the same input).